miio/viomivacuum.py

```diff
@@ -227,96 +223,265 @@ def error(self) -> Optional[str]:
     @property
     def battery(self) -> int:
         """Battery in percentage."""
-        return self.data["battary_life"]
+        return self._data["battary_life"]
 
     @property
     def bin_type(self) -> ViomiBinType:
         """Type of the inserted bin."""
-        return ViomiBinType(self.data["box_type"])
+        return ViomiBinType(self._data["box_type"])
 
     @property
     def clean_time(self) -> timedelta:
         """Cleaning time."""
-        return pretty_seconds(self.data["s_time"])
+        return pretty_seconds(self._data["s_time"])
 
     @property
     def clean_area(self) -> float:
         """Cleaned area in square meters."""
-        return self.data["s_area"]
+        return self._data["s_area"]
 
     @property
     def fanspeed(self) -> ViomiVacuumSpeed:
         """Current fan speed."""
-        return ViomiVacuumSpeed(self.data["suction_grade"])
+        return ViomiVacuumSpeed(self._data["suction_grade"])
 
     @property
     def water_grade(self) -> ViomiWaterGrade:
         """Water grade."""
-        return ViomiWaterGrade(self.data["water_grade"])
+        return ViomiWaterGrade(self._data["water_grade"])
 
     @property
     def remember_map(self) -> bool:
         """True to remember the map."""
-        return bool(self.data["remember_map"])
+        return bool(self._data["remember_map"])
 
     @property
     def has_map(self) -> bool:
         """True if device has map?"""
-        return bool(self.data["has_map"])
+        return bool(self._data["has_map"])
 
     @property
     def has_new_map(self) -> bool:
         """TODO: unknown"""
-        return bool(self.data["has_newmap"])
+        return bool(self._data["has_newmap"])
 
     @property
     def mop_mode(self) -> ViomiMode:
         """Whether mopping is enabled and if so which mode
 
         TODO: is this really the same as mode?
         """
-        return ViomiMode(self.data["is_mop"])
+        return ViomiMode(self._data["is_mop"])
+
+    @property
+    def current_map_id(self) -> float:
+        """Current map id."""
+        return self._data["cur_mapid"]
+
+    @property
+    def hw_info(self) -> str:
+        """Hardware info."""
+        return self._data["hw_info"]
+
+    @property
+    def sw_info(self) -> str:
+        """SoftWare info."""
```

Comment:
```suggestion
        """Software info."""
```

Do you know how this differs from the responses to `miIO.info` call, or is it the same? If the information is the same, I'm wondering if it is worth to have this separately. Same for `hw_info`.

Reply:
Only hardware seems not good with mIO.info

miIO.info
```
Hardware version: Linux
Firmware version: 3.5.3_0047
```

status
```
Hardware version: 1.0.3
Software version: 3.5.3_0047
```

Reply:
What do you want me to do ?

Reply:
I think the extra information is not that useful, so I think simply removing it (or juts keeping the hardware version, if really wanted) is the way to go. I prefer to avoid unnecessary/unhelpful code and keep things like sw/fw information generic for all devices, this will help also if someone wants to use this library to build a generic interface for miio devices.

Reply:
ok, I'll do that

Reply:
Please also remove the `hw_info` as that's already available in the info response, right?

Reply:
I would like to keep it since it's the actual value.

Reply:
As you wish :+1: